# Sales Data Calculator 📊💰

## Task 4 - Level 1 - Day 4

This project analyzes a collection of sales values using Python.

The program calculates:

- Total sales
- Average sales
- Highest sale
- Lowest sale

In [1]:
# Store sales values in a Python list

sales = [
    1200,
    1500,
    980,
    1750,
    2200,
    1450,
    1900,
    1250,
    1600,
    2500
]

print("Sales Values:", sales)

Sales Values: [1200, 1500, 980, 1750, 2200, 1450, 1900, 1250, 1600, 2500]


## 1. Total Sales

The `sum()` function is used to calculate the total sales value.

In [2]:
# Calculate total sales

total_sales = sum(sales)

print("Total Sales:", total_sales)

Total Sales: 16330


## 2. Average Sales

The average sales value is calculated by dividing total sales by the number of sales.

### Formula

Average Sales = Total Sales / Number of Sales

In [3]:
# Calculate average sales

average_sales = total_sales / len(sales)

print("Average Sales:", round(average_sales, 2))

Average Sales: 1633.0


## 3. Highest Sale

The `max()` function is used to find the highest sale value.

In [4]:
# Find the highest sale

highest_sale = max(sales)

print("Highest Sale:", highest_sale)

Highest Sale: 2500


## 4. Lowest Sale

The `min()` function is used to find the lowest sale value.

In [5]:
# Find the lowest sale

lowest_sale = min(sales)

print("Lowest Sale:", lowest_sale)

Lowest Sale: 980


## 5. Final Sales Analysis

The following code displays all calculated sales statistics together.

In [6]:
# Display final sales analysis

print("======================================")
print("          SALES DATA CALCULATOR")
print("======================================")

print("Number of Sales:", len(sales))
print("Total Sales:", total_sales)
print("Average Sales:", round(average_sales, 2))
print("Highest Sale:", highest_sale)
print("Lowest Sale:", lowest_sale)

          SALES DATA CALCULATOR
Number of Sales: 10
Total Sales: 16330
Average Sales: 1633.0
Highest Sale: 2500
Lowest Sale: 980


## 📈 Results

| Statistic | Result |
|---|---:|
| Number of Sales | 10 |
| Total Sales | 16330 |
| Average Sales | 1633 |
| Highest Sale | 2500 |
| Lowest Sale | 980 |

In [7]:
# Display results individually

print("Number of Sales:", len(sales))
print("Total Sales:", total_sales)
print("Average Sales:", round(average_sales, 2))
print("Highest Sale:", highest_sale)
print("Lowest Sale:", lowest_sale)

Number of Sales: 10
Total Sales: 16330
Average Sales: 1633.0
Highest Sale: 2500
Lowest Sale: 980


## 🎤 Interview Questions

### 1. How do you calculate average sales?

Average sales are calculated by dividing total sales by the number of sales.

### 2. Which Python functions are used in this project?

The main functions used are:

- `sum()`
- `max()`
- `min()`
- `len()`

### 3. Why is `len()` used?

`len()` is used to find the total number of sales values in the list.

In [8]:
# Verify the average sales formula

average_check = sum(sales) / len(sales)

print("Average calculated using formula:", average_check)

Average calculated using formula: 1633.0


## 💡 What I Learned

Through this project, I practiced:

- Python lists
- `sum()`
- `max()`
- `min()`
- `len()`
- Average calculation
- Basic numerical data analysis
- Jupyter Notebook
- Git and GitHub

In [9]:
# Final verification

print("Project completed successfully!")
print("Total sales:", total_sales)
print("Average sales:", average_sales)

Project completed successfully!
Total sales: 16330
Average sales: 1633.0
